packages/instrumentation/src/ActiveTracingHelper.ts

```diff
@@ -0,0 +1,127 @@
+import { EngineSpanEvent, ExtendedSpanOptions, SpanCallback, TracingHelper } from '@prisma/internals'
+
+// If true, will publish internal spans as well
+const showAllTraces = process.env.PRISMA_SHOW_ALL_TRACES === 'true'
+
+// https://www.w3.org/TR/trace-context/#examples-of-http-traceparent-headers
+// If traceparent ends with -00 this trace will not be sampled
+// the query engine needs the `10` for the span and trace id otherwise it does not parse this
+const nonSampledTraceParent = `00-10-10-00`
+
+export class ActiveTracingHelper implements TracingHelper {
+  constructor(private traceMiddleware: boolean) {}
+
+  isEnabled(): boolean {
+    return true
+  }
+
+  getTraceParent(context?: Context | undefined): string {
+    const span = trace.getSpanContext(context ?? _context.active())
+    if (span) {
+      return `00-${span.traceId}-${span.spanId}-0${span.traceFlags}`
+    }
+    return nonSampledTraceParent
+  }
+
+  async createSpan(engineSpanEvent: EngineSpanEvent): Promise<void> {
+    // this is only needed for tests and isn't useful otherwise
+    // so that "engine" spans are always emitted after "client"
+    await new Promise((res) => setTimeout(res, 0))
+
+    const tracer = trace.getTracer('prisma') as Tracer
+
+    engineSpanEvent.spans.forEach((engineSpan) => {
+      const spanContext: SpanContext = {
+        traceId: engineSpan.trace_id,
+        spanId: engineSpan.span_id,
+        traceFlags: TraceFlags.SAMPLED,
+      }
+
+      const links = engineSpan.links?.map((link) => {
+        return {
+          context: {
+            traceId: link.trace_id,
+            spanId: link.span_id,
+            traceFlags: TraceFlags.SAMPLED,
+          },
+        }
+      })
+
+      const span = new SpanConstructor(
+        tracer,
+        ROOT_CONTEXT,
+        engineSpan.name,
+        spanContext,
+        SpanKind.INTERNAL,
+        engineSpan.parent_span_id,
+        links,
+        engineSpan.start_time,
+      )
+
+      if (engineSpan.attributes) {
+        span.setAttributes(engineSpan.attributes)
+      }
+
+      span.end(engineSpan.end_time)
+    })
+  }
+
+  getActiveContext(): Context | undefined {
+    return _context.active()
+  }
+
+  runInChildSpan<R>(options: string | ExtendedSpanOptions, callback: SpanCallback<R>): R {
+    if (typeof options === 'string') {
+      options = { name: options }
+    }
+    if (options.internal && !showAllTraces) {
+      return callback()
+    }
+
+    if (options.middleware && !this.traceMiddleware) {
+      return callback()
+    }
+
+    const tracer = trace.getTracer('prisma')
+    const context = options.context ?? this.getActiveContext()
+    const name = `prisma:client:${options.name}`
+    if (options.active === false) {
+      const span = tracer.startSpan(name, options, context)
+      return endSpan(span, callback(span, context))
+    }
+
+    return tracer.startActiveSpan(name, options, (span) => endSpan(span, callback(span, context)))
```

Comment:
Could you please restore the comments from the previous file for this line and line 99?